share/functions/__fish_config_interactive.fish

```diff
@@ -232,6 +232,21 @@ function __fish_config_interactive -d "Initializations that should be performed
     # Load key bindings
     __fish_reload_key_bindings
 
+    if not set -q FISH_UNIT_TESTS_RUNNING
+        # Enable bracketed paste before every prompt (see __fish_shared_bindings for the bindings).
+        # Disable it for unit tests so we don't have to add the sequences to bind.expect
+        function __fish_enable_bracketed_paste --on-event fish_prompt
+            printf "\e[?2004h"
+        end
+
+        # Disable BP before every command because that might not support it.
+        function __fish_disable_bracketed_paste --on-event fish_preexec
+            printf "\e[?2004l"
+        end
+
+        __fish_enable_bracketed_paste
```

Comment:
It's the fact that this isn't loaded until the first `fish_prompt` event has fired that always trips me up. It's counterintuitive. I, and probably most people, expect autoloading this script to happen first. So I'm always surprised when I see this pattern. Hence the need for a comment.